### ISSUE TITLE
Sometimes the player falls under the map when changing instances

### ISSUE THREAD

**Issue by MelonHell** [Minestom/Minestom#2017]
When I change the player's instance sometimes the player ends up in the void and falls down. The world begins to load only when the player moves to an adjacent chunk.

Also, I noticed that in ChunkLoader, the coordinates of the previous instance are passed, which means unnecessary chunks are loaded.

I noticed that this bug occurs more often when using AnvilLoader or another world loader. The flat world generator works very quickly and less frequently triggers this bug.

https://github.com/SkolkovoLab/Minestom/tree/instance-change-bug-demo

https://github.com/Minestom/Minestom/assets/53826469/9a9787c6-cd68-4748-81a4-a2d8c4c5b3cd

**Comment by Spliterash** [comment]
A small addition: I found out that this happens only when the player is near a previous chunk.
As I understand it, this happens because the server does not correctly calculate which chunks the player needs to unload 
In other words, I managed to fix this problem by moving the lobby to 100000 coordinates, and as a result the lobby chunks were very far away from the minigame chunks.

**Comment by domrbeeson** [comment]
I'm experiencing this and can replicate it easily when simulating a slow CPU (docker container with --cpus=0.25) or with a ping above ~200ms

**Comment by ajh123** [comment]
Another way to solve this (but not the most optimal) would use two separate Minestom servers for each instance and use a proxy like Velocity. 

Problems with this solution:
- data synchronisation between instances require plugin messages, or a database (like MySQL or Redis).
- player sees a loading terrain / configuring screen when switching between instances (however, this is common between popular server networks).
- not as lightweight as before.

Benefits of this solution:
- can be more reliable, since each instance is its own server it can be put into its own container with things like Docker, this means if there was a hack on one instance it won't affect all the others.
- can support older versions of Minecraft using plugins like Viaverion or Viabackwards.
- if you need a normal vanilla or fabric server to be accessible you can add it to the proxy.

<hr/>

I've noticed this problem too, so I've decided to use a proxy.

**Comment by TheMeinerLP** [comment]
@MelonHell is this still happend ?

**Comment by Spliterash** [comment]
> [@MelonHell](https://github.com/MelonHell) is this still happend ?

No. To achieve this, we added a check on the block that the player is not allowed inside the block, like on bukkit, and also made teleports with confirmation. After that the problem is gone
